src/record/api.rs

```diff
@@ -170,8 +236,59 @@ pub fn make_map(entries: Vec<(Field, Field)>) -> Map {
   Map { entries: entries }
 }
 
-// TODO: implement `getKeys`, `getValues`, etc., for `Map`.
+/// Trait for type-safe access of an index for a `Map`
+pub trait MapAccessor {
+  fn get_keys<'a>(&'a self) -> Box<ListAccessor + 'a>;
```

Comment:
Now the question is whether or not it is possible to get struct from map value if we return `&[Field]`. In this case we would have to map to `Group(Row)`, and then use one of the accessors. So yes, it may be functionally correct to return `ListAccessor`.

Reply:
Yes, we are only exposing `Row`, `List` and `Map` at the moment, and each of them offers accessors. So, here we need to return `ListAccessor` instead of list of `Field` which is not exposed. 

I'm also hesitant on whether the `RowIter` should return `Box<RowAccessor>` instead of `Row`. In future if we implement a batch-oriented record reader, we may want to return a `Row` from the current `RowBatch`, whose implementation may be different from what we have right now.

Also, it might better to rename the `RowAccessor` to `Row`, and the current `Row` to `RowImpl` or something.

Just some random thought 😀 .